### PR TITLE
PLFM-4120

### PR DESCRIPTION
**Pull request by brucehoff**

### CODE REVIEW

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/AuthorizationManager.java [hunk 1]
```diff
@@ -226,5 +227,15 @@ public AuthorizationStatus canSubscribe(UserInfo userInfo, String objectId,
...
+	
+	/**
+	 * 
+	 * @param userInfo
+	 * @param service
+	 * @param repositoryPath
+	 * @param actionTypes
+	 * @return the permitted actions for the given user on the given repository
+	 */
+	public Set<RegistryEventAction> getPermittedActions(UserInfo userInfo, String service, String repositoryPath, String actionTypes);
```
Comment (brucehoff) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/AuthorizationManager.java:240]: getPermittedDockerRepositoryActions

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/AuthorizationManagerImpl.java [hunk 2]
```diff
@@ -538,4 +551,93 @@ public AuthorizationStatus canSubscribe(UserInfo userInfo, String objectId,
 		}
 		return this.aclDAO.getAccessibleProjectIds(principalIds, ACCESS_TYPE.READ);
 	}
+	
+	/*
+	 * Given a repository path, return a valid parent Id, 
+	 * a project which has been verified to exist. 
+	 * If there is no such valid parent then return null
+	 */
+	public String validParentProjectId(String repositoryPath) {
```
Comment (brucehoff) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/AuthorizationManagerImpl.java:560]: validDockerRepoParentId

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/AuthorizationManagerImpl.java [hunk 3]
```diff
@@ -538,4 +551,93 @@ public AuthorizationStatus canSubscribe(UserInfo userInfo, String objectId,
 		}
 		return this.aclDAO.getAccessibleProjectIds(principalIds, ACCESS_TYPE.READ);
 	}
+	
+	/*
+	 * Given a repository path, return a valid parent Id,
```
Comment (john-hill) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/AuthorizationManagerImpl.java:556]: Docker repo path

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/AuthorizationManagerImpl.java [hunk 4]
```diff
@@ -538,4 +551,93 @@ public AuthorizationStatus canSubscribe(UserInfo userInfo, String objectId,
...
+		}
+		// check that 'repopath' starts with a synapse ID (synID) and synID is a project or folder
+		String parentId;
+		try {
+			parentId = DockerNameUtil.getParentIdFromRepositoryPath(repositoryPath);
+		} catch (IllegalArgumentException e) {
+			return null;
+		}
+
+		Long parentIdAsLong = KeyFactory.stringToKey(parentId);
```
Comment (brucehoff) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/AuthorizationManagerImpl.java:575]: add NodeDao.doesNodeExist(id, type) and use it here.

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/DockerManagerImpl.java [hunk 5]
```diff
@@ -251,7 +148,7 @@ public void dockerRegistryNotification(DockerRegistryEventList registryEvents) {
 				String entityId =  dockerNodeDao.getEntityIdForRepositoryName(repositoryName);
 				if (entityId==null) {
 					// The node doesn't already exist
-					List<EntityHeader> headers = nodeDAO.getEntityHeader(Collections.singleton(KeyFactory.stringToKey(parentId)));
+					List<EntityHeader> headers = nodeDao.getEntityHeader(Collections.singleton(KeyFactory.stringToKey(parentId)));
```
Comment (brucehoff) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/DockerManagerImpl.java:151]: use new NodeDAO method, as above.

**Pull request merged**